sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -483,12 +483,12 @@ public virtual Response<LogsQueryResult> QueryResource(ResourceIdentifier resour
         /// <returns>The logs matching the query.</returns>
         public virtual async Task<Response<LogsQueryResult>> QueryResourceAsync(ResourceIdentifier resourceId, string query, QueryTimeRange timeRange, LogsQueryOptions options = null, CancellationToken cancellationToken = default)
         {
-            string resource = resourceId.ToString();
+            string resource = resourceId.ToString().TrimStart('/');
             using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(LogsQueryClient)}.{nameof(QueryResource)}");
             scope.Start();
             try
             {
-                resource.TrimStart('/');
+                resource = ResourceIdentifier.Parse(resource).Name;
```

Comment:
Sorry, you don't need both the Name property and the Parse call.
You can just do:
```c#
ResourceIdentifier.Parse(resource);
```
I would recommend trimming only after the Parse call.

Reply:
I think we need to trim before because if the user has extra /// in the beginning Parse will throw.

Reply:
We should throw in that case because the resource identifier is invalid.